vms/platformvm/txs/mempool/mempool.go

```diff
@@ -307,6 +306,17 @@ func (m *mempool) deregister(tx *txs.Tx) {
 	m.consumedUTXOs.Difference(inputs)
 }
 
+func (m *mempool) RequestBuildBlock(emptyBlockPermitted bool) {
+	if !m.HasTxs() && !emptyBlockPermitted {
```

Comment:
mega nit - but it's more efficient to short-circuit before the function call rather than after
```suggestion
	if !emptyBlockPermitted && !m.HasTxs() {
```